driver/src/orderbook/streamed/state.rs

```diff
@@ -155,6 +155,61 @@ impl State {
         // order.
         Ok(())
     }
+
+    fn apply_trade(&mut self, event: &Trade, block_batch_id: BatchId) -> Result<()> {
+        self.apply_trade_internal(
+            event.owner,
+            event.order_id,
+            |order| order.trade(event.executed_sell_amount, block_batch_id),
+            |balance| balance.sell(event.executed_sell_amount, block_batch_id),
+            |balance| balance.buy(event.executed_buy_amount, block_batch_id),
+        )
```

Comment:
nit: I think closures can be a nice way to reuse code. Here, I wonder if it would be more straightforward to have a method that returns the mutual references we need (order, sell, buy balance) and then do the apply/revert logic inline?

Reply:
I'm OK either way.

I actually find this to be visually pleasing and easy to read and scan through ("applying a trade for owner and order id where the order applies these trade values and the sell balance applies a sell and buy balance applies a buy").

I also think that returning a tuple with mutable references to the order and balances as you suggest would read in a similar way and would work as well.

My tiny ubernit small suggestion would be to name the parameters to the closures to include which one is the sell token balance and the buy token balance. The reason I suggest this is just so its super clear that these are two distinct token balances and not the same one.

Reply:
I just noticed that I did name the parameters for revert but forgot to do it here. I thought of the "return mutable reference" way to do but slightly preferred this. The other might not actually work because we would be borrowing self mutably multiple times, not sure because I didn't try.